external-resolver/src/test/scala/org/genivi/sota/resolver/test/random/Session.scala

```diff
@@ -20,20 +20,24 @@ object Session {
     for {
       semCmds <- semCommands(sesh.commands)
       _ = println("\n\nCommands: " + sesh.commands.mkString("\n\t", "\n\t", "\n"))
-      semQry  <- semQuery(sesh.query)
-      _ = println("Query: " + sesh.query)
-      _ = println("Result: " + semQry.result + "\n\n")
-    } yield (semCmds :+ semQry)
+      semQrys  <- semQuerys(sesh.querys)
+      qas = sesh.querys zip semQrys
+      _ = qas foreach { case (q, a) =>
+        println("Query: " + q)
+        println("Result: " + a.result + "\n\n")
+      }
+    } yield (semCmds ++ semQrys)
 
   implicit def arbSession(implicit ec: ExecutionContext): Arbitrary[Session] =
     Arbitrary(genSession.runA(initRawStore))
 
   def genSession(implicit ec: ExecutionContext): StateT[Gen, RawStore, Session] =
     for {
-      n    <- lift(Gen.size)
-      cmds <- genCommands(n)
-      qry  <- genQuery
-    } yield Session(cmds, qry)
+      n1   <- lift(Gen.size)
+      cmds <- genCommands(n1) // TODO intermingle queries among commands, to cross-check intermediate states
+      n2    = Math.max(1, n1/2)
+      qrys <- genQuerys(n2)   // TODO why not exercise all endpoint-queries, to compare results with the final RawStore
```

Comment:
Again, because of randomness and because we can run hundreds or thousands runs all endpoints will eventually be queried -- no point in ensuring that they are ALL ALWAYS exercised.